Publish 0.10 release blog (#227)
kserve/website: docs/blog/articles/2023-02-05-KServe-0.10-release.md

# Announcing: KServe v0.10.0

> We are excited to announce KServe 0.10 release, in this release we have been focusing on enabling more KServe networking setup options,

Reviewer: ```suggestion
We are excited to announce KServe 0.10 release. In this release we have been focusing on enabling more KServe networking setup options,
```

> We are excited to announce KServe 0.10 release, in this release we have been focusing on enabling more KServe networking setup options,

Reviewer: ```suggestion
We are excited to announce KServe 0.10 release, in this release we focused on enabling more KServe networking setup options,
```

> improving metrics for supported serving runtimes and increasing support coverage for [Open(aka v2) inference protocol](https://kserve.github.io/website/0.10/modelserving/data_plane/v2_protocol/) for both KServe and ModelMesh.

Reviewer: ```suggestion
improved metrics for supported serving runtimes and increased support coverage for [Open(aka v2) inference protocol](https://kserve.github.io/website/0.10/modelserving/data_plane/v2_protocol/) for both KServe and ModelMesh.
```

> improving metrics for supported serving runtimes and increasing support coverage for [Open(aka v2) inference protocol](https://kserve.github.io/website/0.10/modelserving/data_plane/v2_protocol/) for both KServe and ModelMesh.

Reviewer: model mesh is also kserve component right ? so maybe different wording here - like for single, multi model and modelmesh?

Author: reworded to standard and modelmesh InferenceService

## KServe Networking Options

> Istio is now optional for both `Serverless` and `RawDeployment` mode, please see the [alternative networking guide](https://kserve.github.io/website/0.10/admin/serverless/kourier_networking/) for how you can enable other ingress options supported by Knative with Serverless mode.

Reviewer: ```suggestion
Istio is now optional for both `Serverless` and `RawDeployment` mode. Please see the [alternative networking guide](https://kserve.github.io/website/0.10/admin/serverless/kourier_networking/) for how you can enable other ingress options supported by Knative with Serverless mode.
```

> For Istio users, if you want to turn on full service mesh mode to secure inference services with mutual TLS and enable the traffic policies, please read the [service mesh setup guideline](https://kserve.github.io/website/0.10/admin/serverless/servicemesh/).

Reviewer: should inference services be capital here?

## KServe Telemetry for Serving Runtimes

We have instrumented additional latency metrics in KServe python serving runtimes for `preprocess`, `predict` and `postprocess` handlers.

> In Serverless mode we have extended Knative queue-proxy to enable metrics aggregation for both metrics exposed in `queue-proxy` and `kserve-container`.

Reviewer: ```suggestion
In Serverless mode we have extended Knative queue-proxy to enable metrics aggregation for both metrics exposed in `queue-proxy` and `kserve-container`.  We also added support for exposing metrics from all of our serving runtimes in Serverless Mode. 
```

something like this? we have added support for each serving runtime in serverless mode.

Reviewer: we also have new observability and grafana guide with open source dashboard templates.

        input_tensors = [image_transform(instance) for instance in request.inputs[0].data]
        input_tensors = np.asarray(input_tensors)
        output = self.model(input_tensors)
        torch.nn.functional.softmax(output, dim=1)
        values, top_5 = torch.topk(output, 5)
        result = values.flatten().tolist()
        response_id = generate_uuid()
        infer_output = InferOutput(name="output-0", shape=list(values.shape), datatype="FP32", data=result)
        infer_response = InferResponse(model_name=self.name, infer_outputs=[infer_output], response_id=response_id)
        return infer_response

class CustomTransformer(Model):
    def preprocess(self, request: InferRequest, headers: Dict[str, str]) -> InferRequest:
        input_tensors = [image_transform(instance) for instance in request.inputs[0].data]
        input_tensors = np.asarray(input_tensors)
        infer_inputs = [InferInput(name="INPUT__0", datatype='FP32', shape=list(input_tensors.shape),
                                   data=input_tensors)]
        infer_request = InferRequest(model_name=self.model_name, infer_inputs=infer_inputs)
        return infer_request
```

> You can use the same Python API type `InferRequest` and `InferResponse` for both REST and gRPC protocol, KServe handles the underlying decoding and encoding according to the protocol.

Reviewer: ```suggestion
You can use the same Python API type `InferRequest` and `InferResponse` for both REST and gRPC protocol. KServe handles the underlying decoding and encoding according to the protocol.
```

> New `headers` argument is added to the custom handlers to pass http/gRPC headers or metadata, you can also use this as context dict to pass data between handlers.

Reviewer: ```suggestion
A new `headers` argument is added to the custom handlers to pass http/gRPC headers or metadata. You can also use this as context dict to pass data between handlers.
```

Reviewer: should you also note this is a breaking change

> 

Reviewer: maybe link to serving runtimes web page?

for multiple architectures: `ppc64le`, `arm64`, `amd64`, `s390x`.

## KServe Storage Credentials

- Currently, AWS users need to create a secret with long term/static IAM credentials for downloading models stored in S3.
  Security best practice is to use [IAM role for service account(IRSA)](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) 
  which enables automatic credential rotation and fine-grained access control, see how to [setup IRSA](https://kserve.github.io/website/0.10/modelserving/storage/s3/s3/#create-service-account-with-iam-role).
- Support Azure Blobs with [managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azcli).

## ModelMesh updates
ModelMesh has continued to integrate itself as KServe's multi-model serving backend, introducing improvements and features that better align the two projects. For example, it now supports ClusterServingRuntimes, allowing use of cluster-scoped ServingRuntimes, originally introduced in KServe 0.8.

> Additionally, ModelMesh has introduced support for TorchServe enabling users to serve arbitrary PyTorch models (e.g. eager-mode) in the context of distributed-multi-model serving.

Reviewer: ```suggestion
Additionally, ModelMesh introduced support for TorchServe enabling users to serve arbitrary PyTorch models (e.g. eager-mode) in the context of distributed-multi-model serving.
```

# Announcing: KServe v0.10.0

We are excited to announce KServe 0.10 release, in this release we have been focusing on enabling more KServe networking setup options,
improving metrics for supported serving runtimes and increasing support coverage for [Open(aka v2) inference protocol](https://kserve.github.io/website/0.10/modelserving/data_plane/v2_protocol/) for both KServe and ModelMesh.

## KServe Networking Options

Istio is now optional for both `Serverless` and `RawDeployment` mode, please see the [alternative networking guide](https://kserve.github.io/website/0.10/admin/serverless/kourier_networking/) to see how you can enable other ingress options supported by Knative with Serverless mode.
For Istio users, if you want to turn on full service mesh mode to secure inference services with mutual TLS and enable the traffic policies, please read the [service mesh setup guideline](https://kserve.github.io/website/0.10/admin/serverless/servicemesh/).

## KServe Telemetry for Serving Runtimes

Prometheus metrics are now available for supported serving runtimes including custom python runtimes,

> in Serverless mode we have extended Knative queue-proxy to enable metrics aggregation for both metrics exposed in `queue-proxy` and `kserve-container`. 

Reviewer 2: These two lines may have been intended to be separate sentences.

Please read the [prometheus metrics setup guideline](https://kserve.github.io/website/0.10/modelserving/observability/prometheus_metrics/) for how to enable the metrics scraping and aggregations.

## Open(v2) Inference Protocol Support Coverage

In KServe 0.10, we have added support for Open(v2) inference protocol for KServe custom runtimes, 

> now you can enable v2 REST/gRPC for both custom transformer and predictor with images built by implementing KServe Python SDK API.

Reviewer 2: Same as above. Maybe "...custom runtimes. Now, you can enable..."?

for multiple architectures: `ppc64le`, `arm64`, `amd64`, `s390x`.

## KServe Storage Credentials

- Currently, AWS users need to create a secret with long term/static IAM credentials for downloading models stored in S3.
  Security best practice is to use [IAM role for service account(IRSA)](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) 
  which enables automatic credential rotation and fine-grained access control, see how to [setup IRSA](https://kserve.github.io/website/0.10/modelserving/storage/s3/s3/#create-service-account-with-iam-role).
- Support Azure Blobs with [managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azcli).

## ModelMesh updates

- Support for TorchServe ServingRuntime.
- Support v2 inference protocol for OpenVINO ServingRuntime.
- `ClusterServingRuntime` support for ModelMesh which now works in the same way as KServe.
- Support passing labels and annotations from `ServingRuntimePodSpec` to ModelMesh Pods.

> - Support for `ImagePullSecrets` on ServingRuntime spec.

Reviewer 2: ModelMesh has continued to integrate itself as KServe's multi-model serving backend, introducing improvements and features that better align the two projects. For example, it now supports ClusterServingRuntimes, allowing use of cluster-scoped ServingRuntimes, originally introduced in KServe v0.8. 

Additionally, ModelMesh has introduced support for TorchServe enabling users to serve arbitrary PyTorch models (e.g. eager-mode) in the context of distributed-multi-model serving. 

Other limitations have been addressed as well, such as adding support for BYTES/string type tensors when using the REST inference API for inference requests that require them.
